src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2399,7 +2399,14 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
 
                 if (bailOnFirstError)
                 {
-                    // If we return null here, a fake module info will be created. Since
+                    // If we're trying to load the module, return null so that caches
+                    // are not polluted
+                    if ((manifestProcessingFlags & ManifestProcessingFlags.LoadElements) != 0)
```

Comment:
We should use `HasFlag()` - now it is optimized.

Reply:
Oh didn't see this before! Good idea!